### PR TITLE
fix: Cron schedule

### PR DESCRIPTION
**Pull request by ivasilov**
Small fixes:
- Simplify the implementation in ScheduleSection.
- Fix the schedule input to turn red when its value has error.
- Fix the schedule input to clear the errors when set via presets or AI.
- Fix the ErrorBoundary to handle non-Error errors.
- Minor wording fixes.

### CODE REVIEW

File: apps/studio/components/ui/ErrorBoundaryState.tsx [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
 export const ErrorBoundaryState = ({ error, resetErrorBoundary }: FallbackProps) => {
   const router = useRouter()
-  const message = `Path name: ${router.pathname}\n\n${error.stack}`
-  const isRemoveChildError = error.message.includes("Failed to execute 'removeChild' on 'Node'")
+
+  let errorMessage = ''
+  let urlMessage = ''
+  let isRemoveChildError = false
+  if (isError(error)) {
+    errorMessage = error.message
```
Comment (joshenlim) [inline apps/studio/components/ui/ErrorBoundaryState.tsx:27]: could we use ternary for this? in favor of immutability

`const errorMessage = isError(error) ? error.message ?? ''`

File: apps/studio/components/interfaces/Integrations/CronJobs/CronJobScheduleSection.tsx [hunk 2]
```diff
@@ -76,47 +78,22 @@ export const CronJobScheduleSection = ({ form, supportsSeconds }: CronJobSchedul
...
-        setScheduleString('Invalid cron expression')
-        return
-      }
-
-      setScheduleString(CronToString(schedule))
-    } catch (error) {
-      setScheduleString('Invalid cron expression')
-      console.error('Error converting cron expression to string:', error)
+  const schedule = form.watch('schedule')
+  let scheduleString = ''
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Integrations/CronJobs/CronJobScheduleSection.tsx:89]: hmm can we use ternary for this? we can wrap the `.test` in a separate function that has a try catch

in favor of immutability + consistency

**Comment by joshenlim** [comment]
unrelated - i'm seeing this error in the cron overview page
image

**Comment by joshenlim** [comment]
also seeing this error when i try to generate cron with AI (not sure if its an issue though)
image

File: apps/studio/components/interfaces/Integrations/CronJobs/CronJobScheduleSection.tsx (hunk 2 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/Integrations/CronJobs/CronJobScheduleSection.tsx]: pushed a fix for this!

File: apps/studio/components/interfaces/Integrations/CronJobs/CronJobs.utils.ts [hunk 3]
```diff
@@ -171,5 +174,13 @@ export function getScheduleMessage(scheduleString: string, schedule: string) {
     .map((s, i) => (i === 0 ? s.toLowerCase() : s))
     .join(' ')
 
-  return `The cron will be run ${readableSchedule}.`
+  return `The cron will run ${readableSchedule}.`
+}
+
+export const checkIsSeconds = (value: string) => {
+  try {
+    return secondsPattern.test(value)
```
Comment (ivasilov) [inline apps/studio/components/interfaces/Integrations/CronJobs/CronJobs.utils.ts:182]: This function doesn't throw, `ToCronString` throws.

**Review by joshenlim — APPROVED** [review]
Tested once again on staging - lgtm other than the last console error [here](https://github.com/supabase/supabase/pull/30546#issuecomment-2516921412)! :)

**Pull request merged**